src/middlewares/roleMiddelware.ts

```diff
@@ -0,0 +1,37 @@
+import jwt, { JwtPayload } from 'jsonwebtoken';
+import { Request, Response, NextFunction } from 'express';
+
+import { secret } from '../config/tokenKey.js';
+
+const roleMiddelware = (roles: string) => {
+    return function (req: Request, res: Response, next: NextFunction) {
+        if (req.method === 'OPTIONS') {
+            next();
+        }
+
+        try {
+            let token = req.headers.authorization;
+            if (token) {
+                token = token.split(' ')[1];
+            } else {
+                return res.status(403).json({ message: 'User is not authorized' });
```

Comment:
If the token is invalid or absent then the user is Unauthenticated.
Return `401` status code